lib/DataDrivenLux/src/DataDrivenLux.jl

```diff
@@ -0,0 +1,94 @@
+module DataDrivenLux
+
+using DataDrivenDiffEq
+
+# Load specific (abstract) types
+using DataDrivenDiffEq: AbstractBasis
+using DataDrivenDiffEq: AbstractDataDrivenAlgorithm
+using DataDrivenDiffEq: AbstractDataDrivenResult
+using DataDrivenDiffEq: AbstractDataDrivenProblem
+using DataDrivenDiffEq: DDReturnCode, ABSTRACT_CONT_PROB, ABSTRACT_DISCRETE_PROB
+using DataDrivenDiffEq: InternalDataDrivenProblem
+using DataDrivenDiffEq: is_implicit, is_controlled
+
+using DataDrivenDiffEq.DocStringExtensions
+using DataDrivenDiffEq.CommonSolve
+using DataDrivenDiffEq.CommonSolve: solve!
+using DataDrivenDiffEq.StatsBase
+using DataDrivenDiffEq.Parameters
+using DataDrivenDiffEq.Setfield
+
+using Reexport
+@reexport using Optim
```

Comment:
Not using Optimization? Pure BFGS would be prone to local minima?

Reply:
I'll benchmark later today how this would affect the performance of the optimization routine. 👍

Reply:
I've tested Optimization and the overhead seems a little too much imho


```Julia

fs = (exp,)
arities = (1,)
dag = LayeredDAG(1, 1, 1, arities, fs, skip = true)
X = permutedims(collect(0:0.1:3.0))
Y = sin.(2.0 * X)
@variables x
@parameters p [bounds = (1.0, 2.5), dist = Normal(1.75, 1.0)]
basis = Basis([sin(p * x)], [x], parameters = [p])

dataset = Dataset(X, Y)
rng = StableRNG(2)
candidate = DataDrivenLux.Candidate(rng, dag, basis, dataset)


@btime DataDrivenLux.optimize_candidate!($candidate, $dataset)
# Optim
# 138.708 μs (3231 allocations: 204.42 KiB)
# Optimization
# 1.244 ms (28232 allocations: 2.44 MiB)
```

This constructs the OptimizationFunction once and remakes the problem in the optimize call with initial parameters and the current configuration. Given that this needs to happen for each iteration a few hundred of times, this would slow down the algorithm massively.